Developer-Guide.md

```diff
@@ -0,0 +1,193 @@
+```
+
+The build will create:
+
+- runtime binary: `/usr/local/bin/kata-runtime`
+- configuration file: `/usr/share/defaults/kata-containers/configuration.toml`
+
+# Check hardware requirements
+
+You can now check if your system is capable of creating a Kata Container by running:
+
+```
+$ sudo kata-runtime kata-check
+```
+
+If your system is *not* able to run Kata Containers, this command will error and explain why.
+
+## Enable full debug
+
+Enable full debug as follows:
+```
+$ sudo sed -i -e 's/^# *\(enable_debug\).*=.*$/\1 = true/g' /usr/share/defaults/kata-containers/configuration.toml
+$ sudo sed -i -e 's/^kernel_params = ""/kernel_params = "agent.log=debug"/g' /usr/share/defaults/kata-containers/configuration.toml
+```
+
+# Build and install Kata proxy
+
+```
+$ go get -d -u github.com/kata-containers/proxy
+$ cd $GOPATH/src/github.com/kata-containers/proxy && make && sudo make install
+```
+
+# Build and install Kata shim
+
+```
+$ go get -d -u github.com/kata-containers/shim
+$ cd $GOPATH/src/github.com/kata-containers/shim && make && sudo make install
+```
+
+# Create an image
+
+## Build a custom Kata agent - OPTIONAL
+
+> **Note:**
+>
+> - You only do this step if you wish to test with the latest version of the agent.
+
+```
+$ go get -d -u github.com/kata-containers/agent
+$ cd $GOPATH/src/github.com/kata-containers/agent && make
+```
+
+## Get the osbuilder
+
+```
+$ go get -d -u github.com/kata-containers/osbuilder
+```
+
+## Create a rootfs image
+
+```
+$ cd $GOPATH/src/github.com/kata-containers/osbuilder/rootfs-builder
+$ script -fec 'sudo -E GOPATH=$GOPATH USE_DOCKER=true ./rootfs.sh clearlinux'
+```
+
+> **Note:**
+> 
+> - You must ensure that the *default Docker runtime* is `runc` to make use of the `USE_DOCKER` variable. If that is not the case, simply remove the variable from the command above. See [Checking Docker default runtime](#checking-docker-default-runtime).
+
+### Add a custom agent to the image - OPTIONAL
+
+> **Note:**
+>
+> - You only do this step if you wish to test with the latest version of the agent.
+
+```
+$ sudo install -o root -g root -m 0550 -t rootfs/bin ../../agent/kata-agent
+$ sudo install -o root -g root -m 0440 ../../agent/kata-agent.service rootfs/usr/lib/systemd/system/
+$ sudo install -o root -g root -m 0440 ../../agent/kata-containers.target rootfs/usr/lib/systemd/system/
+```
+
+## Build the image
+
+```
+$ cd $GOPATH/src/github.com/kata-containers/osbuilder/image-builder
+$ script -fec 'sudo -E USE_DOCKER=true ./image_builder.sh ../rootfs-builder/rootfs'
+```
+
+> **Notes:**
+> 
+> - You must ensure that the *default Docker runtime* is `runc` to make use of the `USE_DOCKER` variable. If that is not the case, simply remove the variable from the command above. See [Checking Docker default runtime](#checking-docker-default-runtime).
+> - If you do *not* wish to build under Docker, remove the `USE_DOCKER` variable above and ensure the `qemu-img` command is available on your system.
```

Comment:
- If you do *not* wish to build under Docker, remove the `USE_DOCKER` variable in the previous command and ensure the `qemu-img` command is available on your system.